godot-ffi/src/binding/single_threaded.rs

```diff
@@ -30,13 +38,53 @@ impl BindingStorage {
     #[inline(always)]
     unsafe fn storage() -> &'static Self {
         static BINDING: BindingStorage = BindingStorage {
+            #[cfg(wasm_nothreads)]
+            initialized: Cell::new(false),
+
+            #[cfg(not(wasm_nothreads))]
             main_thread_id: Cell::new(None),
             binding: ManualInitCell::new(),
         };
 
         &BINDING
     }
 
+    /// Marks the binding storage as initialized or deinitialized.
+    /// We store the thread ID to ensure future accesses to the binding only come from the main thread.
+    ///
+    /// # Safety
+    /// Must be called from the main thread. Additionally, the binding storage must be initialized immediately
+    /// after this function if `initialized` is `true`, or deinitialized if it is `false`.
+    ///
+    /// # Panics
+    /// If attempting to deinitialize before initializing, or vice-versa.
```

Comment:
The 2nd part is not checked though.

I think the precondition can be verified outside the `#[cfg]`, e.g. like this:
```rs
if initialized == self.is_initialized() {
    if initialized {
        panic!("already initialized");
    } else {
        panic!("deinitialize without prior initialize");
    }
}
```

Reply:
Interesting, I think this means it wasn't checked before either. So we'll get a free bug fix :)

Reply:
Done! Seems to be working fine.

Reply:
Thanks! Yes, it's always nice to have these drive-by bugfixes :slightly_smiling_face: